public/js/apps/transaction/app.jsx

```diff
@@ -58,14 +58,20 @@ export class TransactionApp extends Component {
     };
   }
 
-  renderChild() {
-    var { app, user, SignIn, Transaction } = this.props;
-    var state = this.state;
-
+  renderError() {
+    var { app } = this.props;
     if (app.error) {
       console.log('rendering app error');
       return <ErrorMessage error={app.error} />;
-    } else if (!state.isValid) {
+    }
+    return null;
+  }
+
+  renderChild() {
+    var { user, SignIn, Transaction } = this.props;
+    var state = this.state;
+
+    if (!state.isValid) {
       // This renders a temporary loading state while we wait for
       // redux to re-render the component with an error to display.
```

Comment:
I guess this comment might be a little misleading now. Perhaps it should just explain how since the component is an error state, nothing else can be rendered.